src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -194,6 +208,7 @@ private async Task<SearchResult> SearchImpl(SearchOptions searchOptions, bool hi
                     long? newContinuationId = null;
                     bool moreResults = false;
                     int matchCount = 0;
+                    DateTime? sortExpr = null;
```

Comment:
This column should not really be called `sortExpr`, because it is not an expression, it is a value. `sortValue` would be better (here and in the SQL).

Also, you should add a comment explaining why this is a `DateTime`.